frugalos_segment/src/test_util.rs

```diff
@@ -0,0 +1,242 @@
+#[cfg(test)]
+pub mod tests {
+    use cannyls::device::{Device, DeviceBuilder, DeviceHandle};
+    use cannyls::lump::LumpId;
+    use cannyls::nvm::MemoryNvm;
+    use cannyls::storage::StorageBuilder;
+    use cannyls_rpc;
+    use cannyls_rpc::DeviceRegistryHandle;
+    use client::storage::StorageClient;
+    use client::Client;
+    use config::*;
+    use fibers::executor::Executor;
+    use fibers::executor::ThreadPoolExecutor;
+    use fibers_global;
+    use fibers_rpc::client::{ClientService, ClientServiceHandle};
+    use fibers_rpc::server::ServerBuilder;
+    use frugalos_raft;
+    use frugalos_raft::{LocalNodeId, NodeId};
+    use futures;
+    use futures::future::Future;
+    use futures::Async;
+    use libfrugalos::entity::device::DeviceId;
+    use slog;
+    use std::net::SocketAddr;
+    use std::thread;
+    use std::time::Duration;
+    use trackable::error::ErrorKindExt;
+    use {Error, ErrorKind, Result};
+    use {Service, ServiceHandle};
+
+    /// Waits for the completion of the given future.
+    pub fn wait<F: Future<Error = Error>>(mut f: F) -> Result<F::Item> {
+        loop {
+            if let Async::Ready(result) = track!(f.poll())? {
+                return Ok(result);
+            }
+
+            thread::sleep(Duration::from_millis(1));
+        }
+    }
+
+    /// Adds `ClusterMember`s to the given cluster.
+    /// この関数は特定のテストシナリオに合わせて作られているので、他のユースケースでは別の関数を作る方がよい。
+    pub fn setup_system(system: &mut System) -> Result<(NodeId, DeviceHandle, StorageClient)> {
+        let (node_id, device_id, device_handle) = system.make_node()?;
+        let mut members = Vec::new();
+
+        members.push(ClusterMember {
+            node: node_id,
+            device: device_id,
+        });
+
+        // Decrements the size of this cluster because we've already created a node.
+        for _ in 0..(system.data_fragments() - 1) {
+            let (node, device, _) = system.make_node()?;
+            members.push(ClusterMember { node, device });
+        }
+
+        let storage_client = system.boot(members)?;
+
+        Ok((node_id, device_handle, storage_client))
+    }
+
+    /// A cluster for testing.
+    /// All implementations under this struct is unstable.
+    pub struct System {
```

Comment:
https://github.com/frugalos/frugalos/blob/28c46838e2a583e2baa5ea54fb83036d12b2bf88/frugalos_raft/src/test_util.rs#L30

I hope to re-use the above code as much as possible but don't have a good idea.